wpimath/src/main/native/include/frc/interpolatable/TimeInterpolatableBuffer.h

```diff
@@ -0,0 +1,89 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <array>
+#include <utility>
+#include <vector>
+#include <functional>
+
+#include <wpi/MathExtras.h>
+
+#include "units/time.h"
+
+namespace frc {
+
+template <typename T>
+class TimeInterpolatableBuffer {
+ public:
+  explicit TimeInterpolatableBuffer(std::function<T(const T&, const T&, double)> func
+    = [](const T& start, const T& end, double t) { return wpi::Lerp(start, end, t); }) :
+    m_interpolatingFunc(func) {}
+
+  void AddSample(units::second_t time, T sample) {
+    // Add the new state into the vector.
+    m_pastSnapshots.emplace_back(time, sample);
+
+    // Remove the oldest snapshot if the vector exceeds our maximum size.
+    while (!m_pastSnapshots.empty()) {
+      const auto& entry = m_pastSnapshots.back();
```

Comment:
This is the most recent entry, not the oldest entry.

Reply:
ah oops I have the order backwords